### PR TITLE
[coqpp] Port plugin to mlg.

### PR DESCRIPTION
**Pull request by ejgallego**
Coq 8.10 will drop support for ml4 files in favor a new pre-processor.

Changes are straightforward but @ppedrot please review the ARGUMENT
EXTEND change.

### CODE REVIEW

File: src/hammer.mlg [hunk 1]
```diff
@@ -791,18 +816,25 @@ let hammer_hook_tac prefix name =
...
 let pr_taclist _ _ _ lst = Pp.pr_comma () (* TODO: LC: I haven't figured out how to print a tactic *)
 
-ARGUMENT EXTEND taclist TYPED AS tactic_list PRINTED BY pr_taclist
-| [ tactic3(tac) "|" taclist(l) ] -> [ tac :: l ]
-| [ tactic3(tac) ] -> [ [ tac ] ]
+let wit_tactic_list = Genarg.ListArg wit_tactic
+
+}
+
+ARGUMENT EXTEND taclist TYPED AS tactic_list PRINTED BY { pr_taclist }
```
Comment (ppedrot) [inline src/hammer.mlg]: You should use `tactic list` instead of defining the genarg by hand, it'll work.

Comment (ejgallego) [reply src/hammer.mlg]: Done thanks.

**Comment by Zimmi48** [comment]
@lukaszcz Your `.ml4` file (future `.mlg` file) is quite big. We usually recommend splitting this in two files: one which contains only the parsing and one which contains the rest of the code (this can be done at any point later on).

**Comment by ejgallego** [comment]
> @lukaszcz Your `.ml4` file (future `.mlg` file) is quite big. We usually recommend splitting this in two files: one which contains only the parsing and one which contains the rest of the code (this can be done at any point later on).

Indeed, in fact I was planning to write this same comment but I forgot, thanks @Zimmi48 

Main disadvantage of a big file is that you lack merlin / tooling support for both ml4 and mlg files. Thus, having only the syntax declarations there is usually more convenient.

**Comment by ejgallego** [comment]
Thanks!

**Pull request merged**